src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -0,0 +1,881 @@
+    {
+      arr[i] = i;
+    }
+  }
+
+  template <typename T>
+  void registerNodalScalarField(const std::string& fieldName)
+  {
+    SLIC_ASSERT_MSG(hasPoints(),
+                    "Cannot register a field with the BlueprintParticleMesh "
+                    "before adding points");
+
+    auto* fld = m_fieldsGroup->createGroup(fieldName);
+    fld->createViewString("association", "vertex");
+    fld->createViewString("topology", m_topoGroup->getName());
+    fld->createViewAndAllocate("values",
+                               sidre::detail::SidreTT<T>::id,
+                               numPoints());
+  }
+
+  template <typename T>
+  void registerNodalVectorField(const std::string& fieldName)
+  {
+    SLIC_ASSERT_MSG(hasPoints(),
+                    "Cannot register a field with the BlueprintParticleMesh "
+                    "before adding points");
+
+    const int SZ = numPoints();
+    const int DIM = dimension();
+
+    auto* fld = m_fieldsGroup->createGroup(fieldName);
+    fld->createViewString("association", "vertex");
+    fld->createViewString("topology", m_topoGroup->getName());
+
+    // create views into a shared buffer for the coordinates, with stride NDIMS
+    auto* buf = m_group->getDataStore()
+                  ->createBuffer(sidre::detail::SidreTT<T>::id, DIM * SZ)
+                  ->allocate();
+    switch(DIM)
+    {
+    case 3:
+      fld->createView("values/x")->attachBuffer(buf)->apply(SZ, 0, DIM);
+      fld->createView("values/y")->attachBuffer(buf)->apply(SZ, 1, DIM);
+      fld->createView("values/z")->attachBuffer(buf)->apply(SZ, 2, DIM);
+      break;
+    case 2:
+      fld->createView("values/x")->attachBuffer(buf)->apply(SZ, 0, DIM);
+      fld->createView("values/y")->attachBuffer(buf)->apply(SZ, 1, DIM);
+      break;
+    default:
+      fld->createView("values/x")->attachBuffer(buf)->apply(SZ, 0, DIM);
+      break;
+    }
+  }
+
+  bool hasField(const std::string& fieldName) const
+  {
+    return m_fieldsGroup->hasGroup(fieldName);
+  }
+
+  template <typename T>
+  axom::ArrayView<T> getNodalScalarField(const std::string& fieldName) const
+  {
+    SLIC_ASSERT_MSG(hasPoints(),
+                    "Cannot extract a field from the BlueprintParticleMesh "
+                    "before adding points");
+
+    T* data = hasField(fieldName)
+      ? static_cast<T*>(
+          m_fieldsGroup->getView(axom::fmt::format("{}/values", fieldName))
+            ->getVoidPtr())
+      : nullptr;
+
+    return axom::ArrayView<T>(data, numPoints());
+  }
+
+  template <typename T>
+  axom::ArrayView<T> getNodalVectorField(const std::string& fieldName) const
+  {
+    SLIC_ASSERT_MSG(hasPoints(),
+                    "Cannot extract a field from the BlueprintParticleMesh "
+                    "before adding points");
+
+    T* data = hasField(fieldName)
+      ? static_cast<T*>(
+          m_fieldsGroup->getView(axom::fmt::format("{}/values/x", fieldName))
+            ->getVoidPtr())
```

Comment:
I don't understand how the `x` values give a vector field.  Please document the assumptions on data storage (such as interleaved, separate, etc.)

Reply:
Thanks! I added a comment about this.